io/StacReader.cpp

```diff
@@ -127,112 +123,206 @@ void StacReader::addArgs(ProgramArgs& args)
+        "items based on.", m_args->collections);
+    args.addSynonym("items", "item_ids");
+    args.addSynonym("catalogs", "catalog_ids");
+    args.addSynonym("collections", "collection_ids");
+
+    //Reader options
+    args.add("validate_schema", "Use JSON schema to validate your STAC objects."
+        " Default: false", m_args->validateSchema, false);
+    args.add("header", "Header fields to forward with HTTP requests",
+        m_args->m_headers);
+    args.add("query", "Query parameters to forward with HTTP requests",
+        m_args->m_query);
+    args.add("reader_args", "Map of reader arguments to their values to pass"
+        " through.", m_args->rawReaderArgs);
+    args.add("requests", "Number of threads for fetching JSON files, Default: 8",
+        m_args->threads, 8);
+    args.addSynonym("requests", "threads");
+
+    //Schema options
     args.add("catalog_schema_url", "URL of catalog schema you'd like to use for"
-        " JSON schema validation.", m_args->catalogSchemaUrl,
+        " JSON schema validation.", m_args->schemaUrls.catalog,
         "https://schemas.stacspec.org/v1.0.0/catalog-spec/json-schema/catalog.json");
+    args.add("collection_schema_url", "URL of collection schema you'd like to use for"
+        " JSON schema validation.", m_args->schemaUrls.collection,
+        "https://schemas.stacspec.org/v1.0.0/collection-spec/json-schema/collection.json");
     args.add("feature_schema_url", "URL of feature schema you'd like to use for"
-        " JSON schema validation.", m_args->featureSchemaUrl,
+        " JSON schema validation.", m_args->schemaUrls.item,
         "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json");
-    args.add("requests", "Number of threads for fetching JSON files, Default: 8",
-        m_args->threads, 8);
-    args.addSynonym("requests", "threads");
 
+}
+
+
+void StacReader::addItem(stac::Item& item)
+{
+    std::string driver = item.driver();
+
+    Stage *stage = m_factory.createStage(driver);
+
+    if (!stage)
+    {
+        std::stringstream msg;
+        msg << "Unable to create driver '" << driver << "' for "
+            << "for asset located at '" << item.assetPath() <<"'";
+        throw pdal_error(msg.str());
+    }
+    Reader* reader = dynamic_cast<Reader*>(stage);
+    if (!reader)
+    {
+        std::stringstream msg;
+        msg << "Unable to cast stage of type '" << driver << "' to a reader";
+        throw pdal_error(msg.str());
+    }
+    reader->setOptions(item.options());
+
+    std::lock_guard<std::mutex> lock(m_p->m_mutex);
+    m_p->m_idList.push_back(item.id());
+    reader->setLog(log());
+    m_merge.setInput(*reader);
+    m_p->m_readerList.push_back(reader);
+}
 
+void StacReader::handleItem(NL::json stacJson, std::string itemPath)
+{
+    stac::Item item(stacJson, m_filename, *m_p->m_connector,
+        m_args->validateSchema);
+    if (item.init(*m_p->m_itemFilters, m_args->rawReaderArgs, m_args->schemaUrls))
```

Comment:
It looks like the missing `else` could represent a failure case - maybe add a comment here about why it's ok to no-op if `item.init()` returns false.